artifactory/commands/replication/template.go

```diff
@@ -0,0 +1,223 @@
+	Create       = "create"
+	Pull         = "pull"
+	Push         = "push"
+
+	// Common replication configuration JSON keys
+	Username               = "username"
+	Password               = "password"
+	URL                    = "url"
+	CronExp                = "cronExp"
+	RepoKey                = "repoKey"
+	EnableEventReplication = "enableEventReplication"
+	Enabled                = "enabled"
+	SyncDeletes            = "syncDeletes"
+	SyncProperties         = "syncProperties"
+	SyncStatistics         = "syncStatistics"
+	PathPrefix             = "pathPrefix"
+	SocketTimeoutMillis    = "socketTimeoutMillis"
+)
+
+type ReplicationTemplateCommand struct {
+	path string
+}
+
+func NewReplicationTemplateCommand() *ReplicationTemplateCommand {
+	return &ReplicationTemplateCommand{}
+}
+
+func (rtc *ReplicationTemplateCommand) SetTemplatePath(path string) *ReplicationTemplateCommand {
+	rtc.path = path
+	return rtc
+}
+
+func (rtc *ReplicationTemplateCommand) CommandName() string {
+	return "rt_replication_template"
+}
+
+func (rtc *ReplicationTemplateCommand) RtDetails() (*config.ArtifactoryDetails, error) {
+	// Since it's a local command, usage won't be reported.
+	return nil, nil
+}
+
+func (rtc *ReplicationTemplateCommand) Run() (err error) {
+	replicationTemplateQuestionnaire := &utils.InteractiveQuestionnaire{
+		MandatoryQuestionsKeys: []string{JobType, RepoKey},
+		QuestionsMap:           questionMap,
+	}
+	err = replicationTemplateQuestionnaire.Perform()
+	if err != nil {
+		return err
+	}
+	resBytes, err := json.Marshal(replicationTemplateQuestionnaire.AnswersMap)
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+	if err = ioutil.WriteFile(rtc.path, resBytes, 0644); err != nil {
+		return errorutils.CheckError(err)
+	}
+	log.Info(fmt.Sprintf("Replication creation config template successfully created at %s.", rtc.path))
+	return nil
+}
+
+var questionMap = map[string]utils.QuestionInfo{
+	utils.OptionalKey: {
+		Msg:          "Select the next property",
+		PromptPrefix: ">",
+		AllowVars:    false,
+		Writer:       nil,
+		MapKey:       "",
+		Callback:     optionalKeyCallback,
+	},
+	TemplateType: {
+		Options: []prompt.Suggest{
+			{Text: Create, Description: "Template for creating a new replication job"},
+		},
+		Msg:          "Select the template type",
+		PromptPrefix: ">",
+		AllowVars:    true,
+		Writer:       nil,
+		MapKey:       "",
+		Callback:     nil,
+	},
+	JobType: {
+		Options: []prompt.Suggest{
+			{Text: Pull, Description: "Pull replication job"},
+			{Text: Push, Description: "Push replication job"},
+		},
+		Msg:          "Select job type",
+		PromptPrefix: ">",
+		AllowVars:    true,
+		Writer:       utils.WriteStringAnswer,
+		MapKey:       JobType,
+		Callback:     jobTypeCallback,
+	},
+	RepoKey: {
+		Msg:          "Enter repo key",
+		PromptPrefix: ">",
+		AllowVars:    true,
+		Writer:       utils.WriteStringAnswer,
+		MapKey:       RepoKey,
+		Callback:     nil,
+	},
+	Username: {
```

Comment:
We should not ask for a server URL, username and password, but for a server ID instead, The template should include the server ID only.

Reply:
Done